errors/next-prerender-data.md

```diff
@@ -0,0 +1,170 @@
+---
+
+#### Why This Error Occurred
+
+When the experimental flag `dynamicIO` is enabled Next.js expects you to explicitly describe whether data accessed during render should be evaluated ahead of time, while prerendering, or at Request time while rendering.
+
+Data in this context refers to both reading from the Request using Next.js built-in Request functions like `cookies()`, `headers()`, `draftMode()`, and `connection()` functions, Next.js built-in Request props `params`, and `searchParams`, as well as any asynchronous data fetching technique such as `fetch()` or other network request library or database clients and more.
+
+By default, any data accessed during render is treated as if it should be evaluated at Request time. To explicitly communicate to Next.js that some data should be prerenderable you must explicitly cache it using `"use cache"` or `unstable_cache`.
+
+However, you may have taken great care to have a fully or partially prerenderable route and it would be quite easy to accidentally make such a route non-prerenderable by introducing a new data dependency you forgot to cache. To prevent this Next.js requires that data accessed without caching must be inside a Suspense boundary that defines a fallback UI to use while loading this data.
+
+This makes React's `Suspense` component an explicit opt-in to allowing uncached data access.
+
+To ensure you have a fully prerenerable route you should omit any Suspense boundaries in your route. Suspense is useful for loading UI dynamically but if you have entirely prerenderable pages there is no need to have fallback UI because the primary UI will always be available.
+
+To allow uncached data anywhere in your application you can add a Suspense boundary just inside your `<body>` tag in your Root Layout. We don't recommend you do this however because you will likely want to scope Suspense boundaries around more granular component boundaries that provide fallback UI specific to individual Components.
+
+Hybrid applications will typically use a combination of both techniques, with your top level shared Layouts avoiding Suspense so they can be prerendered for static pages and your layouts that actually have data dependencies will define fallback UI.
+
+Note: While external data can be accessed inside `"use cache"` and `unstable_cache()`, Request data such as `cookies()` cannot because we don't know about cookies before a Request actually occurs. If your application needs to read cookies the only recourse you have is to opt into allowing this data read using `Suspense`.
+
+#### Possible Ways to Fix It
+
+If you are accessing external data that doesn't change often and your use case can tolerate stale results while revalidating the data after it gets too old you should wrap the data fetch in a `"use cache"` function. This will instruct Next.js to cache this data and allow it to be accessed without defining a fallback UI for the component that is accessing this data.
```

Comment:
```suggestion
If you are accessing external data that doesn't change often and your use case can tolerate stale results while revalidating the data after it gets too old, you should wrap the data fetch in a `"use cache"` function. This will instruct Next.js to cache this data and allow it to be accessed without defining a fallback UI for the component that is accessing this data.
```